src/artifact-downloader.ts

```diff
@@ -0,0 +1,357 @@
+import { createHash } from 'node:crypto'
+
+import { createHeliaHTTP } from '@helia/http'
+import type { unixfs } from '@helia/unixfs'
+import { unixfs as createUnixfs } from '@helia/unixfs'
+import axios from 'axios'
+import debug from 'debug'
+import { CID } from 'multiformats/cid'
+
+import { decompressIfNeeded } from './brotli-decompress.js'
+import type { Artifact, ArtifactHashesJson } from './definitions.js'
+import { ArtifactName, RAILGUN_ARTIFACTS_CID_POI, RAILGUN_ARTIFACTS_CID_ROOT } from './definitions.js'
+import ARTIFACT_V2_HASHES from './json/artifact-v2-hashes.json'
+import { getPathForPOIArtifact, isPOIArtifact } from './poi-artifact-downloader.js'
+import { isDefined } from './utils.js'
+
+const dbg = debug('artifact-fetcher:downloader')
+
+type HeliaNode = Awaited<ReturnType<typeof createHeliaHTTP>>
+
+let heliaNode: HeliaNode | undefined
+let fs: ReturnType<typeof unixfs> | undefined
+
+/**
+ * Initializes the Helia node and UnixFS API if not already initialized.
+ * @returns A promise that resolves to an object containing the Helia node and UnixFS API.
+ */
+async function initHelia () {
+  if (heliaNode) {
+    return { heliaNode, fs }
+  }
+
+  // Create Helia with optimized block brokers
+  heliaNode = await createHeliaHTTP()
```

Comment:
Will add stop to node instance to prevent from persisting

Reply:
add todo